src/main/java/com/example/onlinebookstore/controller/BookController.java

```diff
@@ -16,36 +22,45 @@
 import org.springframework.web.bind.annotation.ResponseStatus;
 import org.springframework.web.bind.annotation.RestController;
 
+@Tag(name = "Book management", description = "Endpoints for managing books")
 @RestController
 @RequestMapping("/api/books")
 @RequiredArgsConstructor
+@Validated
 public class BookController {
     private final BookService bookService;
-    
+
+    @Operation(summary = "Get all books", description = "Get a list of all books")
     @GetMapping
-    public List<BookDto> getAll() {
-        return bookService.findAll();
+    public List<BookDto> getAll(Pageable pageable) {
```

Comment:
you may also use `@PageableDefault` annotation here

if you find that useful of course :)